core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestUpgradeAction.java

```diff
@@ -58,32 +57,32 @@ public RestUpgradeAction(Settings settings, RestController controller) {
     }
 
     @Override
-    public Runnable prepareRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         if (request.method().equals(RestRequest.Method.GET)) {
-            return handleGet(request, channel, client);
+            return handleGet(request, client);
         } else if (request.method().equals(RestRequest.Method.POST)) {
-            return handlePost(request, channel, client);
+            return handlePost(request, client);
         }
-        return () -> {};
+        return channel -> {};
```

Comment:
Rather an `IllegalArgumentException`; `UnsupportedOperationException` means "there are no parameters you can pass here to make this method execute` but `IllegalArgumentException` says "there are parameters you can pass here to make this method execute, just not the parameters that you gave me". I get your point though, and will push a commit. (Also, this is in should-never-happen territory, since the handler will not be invoked without the method being `GET` or `POST`.)